src/core_modules/capture-core/reducers/descriptions/app.reducerDescriptionGetter.js

```diff
@@ -131,8 +131,8 @@ export const getAppReducerDesc = (appUpdaters: Updaters) => createReducerDescrip
         ...state,
         locationSwitchInProgress: true,
     }),
-    [lockedSelectorActionTypes.FROM_URL_UPDATE]: (state, action) => ({
-        ...state,
+    [lockedSelectorActionTypes.FROM_URL_UPDATE]: (state, action) => ({ ...state,
```

Comment:
Hey,

IMHO we should avoid adding more logic to the deprecated `LockedSelector` epics and reducers. I know most pages still use this logic to validate the URL parameters and hydrate `currentSelection` and `app` with data. As far as I know, we want to get rid of it soon and let each page handle its own initializations. There are pages where all the `LockedSelector` logic was already removed like the `/enrollment` and `/enrollmentEventEdit`. In these pages, the current feature implementation is not working. 

My proposal is to handle this new feature inside `OrgUnitField` and its parent components (`SingleOrgUnitSelectField` and `ScopeSelector/QuickSelector/OrgUnitSelector`). When the org unit is deselected from within the parents they can store the `previousOrgUnitId` in their state. After the deselection, the stored `previousOrgUnitId` can be passed down to `OrgUnitField` which can use the `usePreviousOrganizationUnit` hook to get the path.

Let me know if this makes sense and if the design gets complicated maybe we can discuss it more in our FE check-in meetings.

Thank you!

Reply:
hey @simonadomnisoru, thank you for your suggestion, I thought about setting the previous id when resetting the orgUnitId but the component is rerendered so the state is empty, so I have to based on the state in redux, currently there are still a lot of logic in `currentSelections` and `LockedSelector`, I know it's not the good idea to add the new logic into that but currently have no idea how that could work without relying on redux. 
Please let me know if that's clear or we can set up separate meeting. Thanks